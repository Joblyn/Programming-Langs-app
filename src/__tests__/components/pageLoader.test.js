import React from "react";
import { render, cleanup } from "@testing-library/react";
import PageLoader from "../../components/pageloader/pageloader";

describe("<PageLoader />", () => {
  afterEach(cleanup);

  // UI test
  it("renders component correctly", () => {
    const { container } = render(<PageLoader />);

    expect(container.getElementsByClassName("spinner").length).toBe(1);
    expect(container.getElementsByClassName("spinner-item").length).toBe(3);
    expect(container.firstChild).toMatchSnapshot();
  });
});
