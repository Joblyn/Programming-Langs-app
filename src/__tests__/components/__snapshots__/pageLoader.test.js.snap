// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageLoader /> renders component correctly 1`] = `
<div
  class="h-screen flex justify-center items-center"
>
  <div
    class="spinner"
  >
    <div
      class="spinner-item"
    />
    <div
      class="spinner-item"
    />
    <div
      class="spinner-item"
    />
  </div>
</div>
`;
